util/fhir.ts

```diff
@@ -34,6 +35,36 @@ export function createPatientResourceString(birthDate: string): string {
   return JSON.stringify(pt, null, 2);
 }
 
+/**
+ * Identifies the primary code path of a resource and constructs a string which displays
+ * resource summary information depending on what is a available
+ * @param resource {Object} a fhir Resource object
+ * @returns {String} displaying the code and display text, code, or id of the resource or nothing
+ */
+export function getFhirResourceSummary(resource: fhir4.Resource) {
+  const primaryCodePath = parsedPrimaryCodePaths[resource.resourceType]?.primaryCodePath;
+  const resourceCode = `${fhirpath.evaluate(resource, `${primaryCodePath}.coding.code`)}`;
+  const resourceDisplay = `${fhirpath.evaluate(resource, `${primaryCodePath}.coding.display`)}`;
+
+  if (primaryCodePath) {
+    if (resourceCode !== '' && resourceDisplay !== '') {
+      return `(${resourceCode}: ${resourceDisplay}`;
+    } else if (resourceCode !== '') {
+      return `(${resourceCode})`;
+    } else if (resourceDisplay !== '') {
+      return `(${resourceDisplay})`;
+    } else if (fhirpath.evaluate(resource, 'id')) {
+      return `(${fhirpath.evaluate(resource, 'id')})`;
+    } else {
+      return '';
+    }
+  } else if (fhirpath.evaluate(resource, 'id')) {
+    return `(${fhirpath.evaluate(resource, 'id')})`;
+  } else {
+    return '';
+  }
+}
```

Comment:
```suggestion
    }
  }
  
  if (resource.id)) {
    return `(${resource.id})`;
  } else {
    return '';
  }
}
```

To not repeat the grabbing of the id and making that string, let the situation where we cant find a code fall out of the top level if block and then get picked up by the code at the end. Also fhirpath isn't necessary for grabbing the id.